cannon/mipsevm/instrumented_test.go

```diff
@@ -0,0 +1,139 @@
+package mipsevm
```

Comment:
`InstrumentedState` should be in the `single_threaded` package as it's specific to that. As I imagine we'll have a similar type for `multithreaded` in the future.
Doing this breaks the circular dependency and we can replace the tests back to the way they were.

Reply:
Yeah, that makes sense - moved these.  I left the tests split out though because state_test.go really seems to be more focused on unit-testing stage.go whereas these tests I isolated are more testing the mips / InstrumentedState step logic.